packages/cozy-sharing/src/components/ShareDialogCozyToCozy.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react'
+import { FixedDialog } from 'cozy-ui/transpiled/react/CozyDialogs'
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+
+import { default as DumbShareByLink } from './ShareByLink'
+import { default as DumbShareByEmail } from './ShareByEmail'
+import WhoHasAccess from './WhoHasAccess'
+
+import cx from 'classnames'
+import styles from '../share.styl'
+const ShareDialogCozyToCozy = ({
+  onClose,
+  documentType,
+  document,
+  permissions,
+  link,
+  onShareByLink,
+  onRevokeLink,
+  onUpdateShareLinkPermissions,
+  showShareOnlyByLink,
+  showShareByEmail,
+  hasSharedParent,
+  recipients,
+  sharingDesc,
+  contacts,
+  groups,
+  createContact,
+  onShare,
+  needsContactsPermission,
+  sharing,
+  showWhoHasAccess,
+  isOwner,
+  onRevoke,
+  onRevokeSelf
+}) => {
```

Comment:
It seems like a smell to have so much props :/

Reply:
I think the previous goal was to have "Dumb" components. I don't know why exactly but I think it is done like that to do not have to mock anything in our test to only pass the wanted props. 

This DumbComponent, is called in context of... a SharingConsumer. For instance: https://github.com/cozy/cozy-libs/blob/master/packages/cozy-sharing/src/index.jsx#L509-L521 and then called itself in... an other SharingConsumer: https://github.com/cozy/cozy-libs/blob/master/packages/cozy-sharing/src/components/EditableSharingModal.jsx#L21-L36

Today there is no plan to change this stuff, I am open to all ideas to improve this code. However, I'm going to create a kind of playground to make it easier to refacto and make sure that nothing breaks.

Reply:
> I am open to all ideas to improve this code. 

```jsx
 <DumbShareModal
    // can come from queryConnect or a redux selector
    contacts={contacts}

    // Not necessary
    createContact={contact => client.create(Contact.doctype, contact)}

    // Necessary
    document={document}

    // Can it be removed in favor of be document._type ? 
    documentType={documentType}


    groups={groups}
    
    // hasSharedChild could have the condition if (!documentPath) { return false } and then we can remove this prop
    hasSharedChild={documentPath && hasSharedChild(documentPath)}

    // hasSharedParent could have the condition if (!documentPath) { return false } and then we can remove this prop
    hasSharedParent={documentPath && hasSharedParent(documentPath)}

    // Not necessary, this should be done by the component
    isOwner={isOwner(document.id)}

    // Not necessary, this should be done by the component
    link={getSharingLink(document.id)}
    onRevoke={revoke}
    onRevokeLink={revokeSharingLink}
    onRevokeSelf={revokeSelf}
    onShare={share}
    onShareByLink={shareByLink}
    onUpdateShareLinkPermissions={updateDocumentPermissions}

    // Not necessary, can be done by the component
    permissions={getDocumentPermissions(document.id)}
  
    // Not necessary, can be done by the component
    recipients={getRecipients(document.id)}

    // Not necessary, can be done by the component
    sharing={getSharingForSelf(document.id)} />


 <DumbShareModal
    document={document}
    link={getSharingLink(document.id)}
    onRevoke={revoke}
    onRevokeLink={revokeSharingLink}
    onRevokeSelf={revokeSelf}
    onShare={share}
    onShareByLink={shareByLink}
    onUpdateShareLinkPermissions={updateDocumentPermissions} />
```

We could pair on something like that :)